src/swa.d.ts

```diff
@@ -91,3 +91,30 @@ declare type UserDefinedRoute = {
   rewrite?: string;
   redirect?: string;
 };
+
+declare type UserDefinedResponseOverride = {
+  statusCode?: number;
+  redirect?: string;
+  rewrite?: string;
+};
+
+declare type BadRequestOverride = {
+  "400": UserDefinedResponseOverride;
+};
+declare type UnauthorizedOverride = {
+  "401": UserDefinedResponseOverride;
+};
+declare type ForbiddenOverride = {
+  "403": UserDefinedResponseOverride;
+};
+declare type NotFoundOverride = {
+  "404": UserDefinedResponseOverride;
+};
+
+declare type UserConfig = {
+  userDefinedRoutes: UserDefinedRoute[];
+  userDefinedResponseOverrides?: Record<BadRequestOverride | UnauthorizedOverride | ForbiddenOverride | NotFoundOverride>;
+  navigationFallback?: Record<string, string>;
+  globalHeaders?: Record<string, string>;
+  mimeTypes?: Record<string, string>;
+};
```

Comment:
Added types for UserConfig
TODO: Currently only userDefinedRoutes and userDefinedResponseOverrides are handles in the application. The other three types need to be implemented and tested.